### PR TITLE
[fix-commission-petition] fix commission test utility function

### PR DESCRIPTION
**Pull request by gjc13**
fix reject handling and commissioner id comparing

### CODE REVIEW

File: src/common/tlv.hpp [hunk 1]
```diff
@@ -196,6 +196,13 @@ enum
     kJoinerRouterKek         = 21,
 };
 
+enum
+{
+    kPetitionAccepted = 1,
```
Comment (bukepo) [inline src/common/tlv.hpp]: name as `kState*`?

File: tests/meshcop/commissioner.cpp [hunk 2]
```diff
@@ -838,7 +832,7 @@ int CommissionerServe(Context &aContext)
 exit:
     otbrLog(OTBR_LOG_INFO, "CommissionerServe: result=%d", ret);
 
-    return ret;
+    return aContext.mState == kStateDone ? 0 : -1;
```
Comment (librasungirl) [inline tests/meshcop/commissioner.cpp]: why not return `ret` directly?

Comment (gjc13) [reply tests/meshcop/commissioner.cpp]: Returning ret directly will need additional flags for the double loop

Comment (librasungirl) [reply tests/meshcop/commissioner.cpp]: A little confused about the function of the variable `ret` in your interpretation.
How about initializing `ret` as `-1 `and exit without reassign `ret` with `errno` when fails to create the socket in Line 746?

**Comment by gjc13** [comment]
No need to merge now since this part of code is undergoing a full rewrite.

**Pull request closed without merging**